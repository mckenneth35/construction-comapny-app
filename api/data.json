[
  {
    "id": 1,
    "name": "Albert Werner Excavation",
    "logoUrl": "https://placekitten.com/100/100",
    "specialties": [
      "EXCAVATION"
    ],
    "city": "Berlin"
  },
  {
    "id": 2,
    "name": "Newton Einstein Edison Mozart da Vinchi McDonald Excavation",
    "logoUrl": "https://placekitten.com/200/200",
    "specialties": [
      "EXCAVATION"
    ],
    "city": "London"
  },
  {
    "id": 3,
    "name": "GWS Construction",
    "logoUrl": "https://placekitten.com/200/250",
    "specialties": [
      "EXCAVATION",
      "PLUMBING",
      "ELECTRICAL",
      "FIRE_SAFETY"
    ],
    "city": "Berlin"
  },
  {
    "id": 4,
    "name": "Klusenburg IR Pro",
    "logoUrl": "https://placekitten.com/220/200",
    "specialties": [
      "EXCAVATION",
      "PLUMBING"
    ],
    "city": "Lens"
  },
  {
    "id": 5,
    "name": "Meyer & Partner Construction",
    "logoUrl": "https://placekitten.com/520/530",
    "specialties": [
      "EXCAVATION",
      "ELECTRICAL"    ],
    "city": "Santa Claus Village Joulupukin Pajakyl√§"
  },
  {
    "id": 6,
    "name": "Nother-Bau",
    "logoUrl": "https://placekitten.com/200/400",
    "specialties": [
      "PLUMBING"
    ],
    "city": "Munich"
  },
  {
    "id": 7,
    "name": "Prignitz",
    "logoUrl": "https://placekitten.com/400/400",
    "specialties": [
      "FIRE_SAFETY"
    ],
    "city": "Antwerp"
  },
  {
    "id": 8,
    "name": "Sommer Road Building",
    "logoUrl": "https://placekitten.com/400/200",
    "specialties": [
      "EXCAVATION",
      "PLUMBING",
      "ELECTRICAL"
    ],
    "city": "Geneva"
  },
  {
    "id": 9,
    "name": "Buenos Dias Construct",
    "logoUrl": "https://placekitten.com/500/200",
    "specialties": [
      "EXCAVATION",
      "FIRE_SAFETY"
    ],
    "city": "London"
  },
  {
    "id": 10,
    "name": "Marco Reus Electric",
    "logoUrl": "https://placekitten.com/450/360",
    "specialties": [
      "ELECTRICAL"
    ],
    "city": "Dortmund"
  },
  {
    "id": 11,
    "name": "All Rounder Co",
    "logoUrl": "https://placekitten.com/350/300",
    "specialties": [
      "EXCAVATION",
      "PLUMBING",
      "ELECTRICAL",
      "FIRE_SAFETY"
    ],
    "city": "Paris"
  },
  {
    "id": 12,
    "name": "Musical Construction",
    "logoUrl": "https://placekitten.com/500/700",
    "specialties": [
      "PLUMBING",
      "ELECTRICAL"
    ],
    "city": "Vienna"
  },
  {
    "id": 13,
    "name": "Wenger & Wagner & Werner Excavation",
    "logoUrl": "https://placekitten.com/60/50",
    "specialties": [
      "EXCAVATION",
      "ELECTRICAL"
    ],
    "city": "Berlin"
  },
  {
    "id": 14,
    "name": "Alpha Wenger Excavation",
    "logoUrl": "https://placekitten.com/500/500",
    "specialties": [
      "PLUMBING",
      "ELECTRICAL",
      "FIRE_SAFETY"
    ],
    "city": "Berlin"
  },
  {
    "id": 15,
    "name": "Albert Wagner Fire Safety",
    "logoUrl": "https://placekitten.com/60/70",
    "specialties": [
      "FIRE_SAFETY"
    ],
    "city": "Berlin"
  }
]