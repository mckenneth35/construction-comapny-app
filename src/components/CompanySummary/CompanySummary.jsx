import React, { useState } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

import Company from '../../types/Company';
import { SPECIALTY_TYPES_NAME, SPECIALTY_TYPES_TAG_COLOR_THEME } from '../../constants/specialties';

const CompanySummary = ({
  company,
}) => {
  const [hovering, setHovering] = useState(false);

  return (
    <div
      onMouseOver={() => setHovering(true)}
      onMouseLeave={() => setHovering(false)}
      className={clsx('company-summary', {
        'company-summary-hover': hovering,
      })}
    >
      <div className='company-summary-img'>
        <img
          className='logo-img'
          src={company.logoUrl}
        />
      </div>

      <div id='Company-summary-name' className='company-summary-name cut-text'>
        {company.name}
      </div>

      <div className='company-summary-specialties'>
        <span id='Specialty-tags' className='flexwrap'>
          {company?.specialties?.map((spec) => (
            <div id='Specialty-tag' className={clsx('specialty-tag', SPECIALTY_TYPES_TAG_COLOR_THEME[spec])} key={spec}>
              {SPECIALTY_TYPES_NAME[spec]}
            </div>
          ))}
        </span>
      </div>

      <div className='company-summary-city cut-text'>
        {company.city}
      </div>
    </div>
  );
};

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
CompanySummary.propTypes = {
  company: PropTypes.exact(Company),
};


export default CompanySummary;
