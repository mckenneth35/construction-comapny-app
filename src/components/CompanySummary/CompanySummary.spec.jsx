import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CompanySummary from './CompanySummary';
import { mockCompanies } from '../../tests/data/Companies';

describe('component:CompanySummary', () => {
  it('Should render Company Summary', () => {
    const wrapper = shallow(
      <CompanySummary
        company={mockCompanies[0]}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
