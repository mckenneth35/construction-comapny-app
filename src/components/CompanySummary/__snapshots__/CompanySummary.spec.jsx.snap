// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:CompanySummary Should render Company Summary 1`] = `
<div
  className="company-summary"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="company-summary-img"
  >
    <img
      className="logo-img"
      src="https://placekitten.com/220/250"
    />
  </div>
  <div
    className="company-summary-name cut-text"
    id="Company-summary-name"
  >
    Jumbo EX
  </div>
  <div
    className="company-summary-specialties"
  >
    <span
      className="flexwrap"
      id="Specialty-tags"
    >
      <div
        className="specialty-tag grey-theme"
        id="Specialty-tag"
        key="EXCAVATION"
      >
        Excavation
      </div>
      <div
        className="specialty-tag blue-theme"
        id="Specialty-tag"
        key="PLUMBING"
      >
        Plumbing
      </div>
      <div
        className="specialty-tag yellow-theme"
        id="Specialty-tag"
        key="ELECTRICAL"
      >
        Electrical
      </div>
      <div
        className="specialty-tag orange-theme"
        id="Specialty-tag"
        key="FIRE_SAFETY"
      >
        Fire Safety
      </div>
    </span>
  </div>
  <div
    className="company-summary-city cut-text"
  >
    Erster Schweizer Stadt
  </div>
</div>
`;
